### PR TITLE
Contributing

### PR DESCRIPTION
**Pull request by mo-marqh**
This PR adds a Contributing guide to the repository, linked to the repostories.md detailed advice.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Contribution Guidelines for TCD XIOS Demonstration
+
+For code contributions, please create your own Fork of this repository and propose code via Pull Request, as detailed in [repositories guidance](repositories.md)
+
+All code contributions will require a code review prior to merging to `main`.
+
+Pull Requests should test your code contributions with respect to several versions of XIOS, and provide a report within the PR. All tests must pass for merging to be allowed.
+Please ensure that all new directories have an `__init__.py` to ensure they are 'discovered' by Python's `unittest` and suitable tests of the contributed code.
+
+## Licensing
```
Comment (TeranIvy) [inline CONTRIBUTING.md:10]: Blank line below the title?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+By contributing code to this repository, You hereby grant to the Met Office a perpetual, worldwide, non-exclusive, royalty-free, irrevocable copyright license to reproduce, prepare derivative works of, publicly display, publicly perform, sublicense, and distribute Your Contributions and such derivative works under the terms of the license specified  for this repository.
```
Comment (TeranIvy) [inline CONTRIBUTING.md:11]: Is this a license text used in existing MO codes?

File: LICENSE [hunk 3]
```diff
@@ -1,28 +1,506 @@
...
-IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
-DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
-FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
-DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
-SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
-CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
-OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
-OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+CeCILL FREE SOFTWARE LICENSE AGREEMENT
```
Comment (TeranIvy) [inline LICENSE:2]: I think it is worth checking with Glenn whether MO codebases can use CeCILL license.

**Review by TeranIvy — DISMISSED** [review]
This sounds good, thank you! I think it should go to Harry for review as well, and there is a note to check for license.

File: LICENSE (hunk 3 above)
Comment (mo-marqh) [reply LICENSE]: removed

File: CONTRIBUTING.md (hunk 2 above)
Comment (mo-marqh) [reply CONTRIBUTING.md]: I've updated this, using the pattern applied within the Rose code base as a good practice guide

**Review by harry-shepherd — APPROVED** [review]
Hi Mark, this looks great.

**Comment by harry-shepherd** [comment]
As Iva's comments have been addressed, I have merged this change.

**Pull request merged**